lib/maplib/map.h

```diff
@@ -0,0 +1,234 @@
+/*
+	This file is part of Warzone 2100.
+	Copyright (C) 1999-2004  Eidos Interactive
+	Copyright (C) 2005-2021  Warzone 2100 Project
+
+	Warzone 2100 is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	Warzone 2100 is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with Warzone 2100; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#pragma once
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "map_debug.h"
+#include "map_io.h"
+#include "terrain_type.h"
+#include "map_terrain_types.h"
+
+// MARK: - Various defines needed by both maplib and the game's map-handling code
+
+#define MAP_MAXWIDTH	256
```

Comment:
I don't like `#define`s. Macros are blindly inserted by preprocessor, can be hard to debug and generally generate more problems than they solve. If we want to write more modern C++, in new code we should use `const` or `constexpr`. `const int` init in headers is allowed even in C++11, for other consts we might need to wait for C++14 or 17

Reply:
Yep - this is just stage 1. (Those bits at the top were just blindly moved from the old WZ `map.h` to maplib's `map.h`.)

Once we move to C++14, there's a *bunch* of these old bits that'll be nice to revise.